tests/xtro-sharpie/EnumCheck.cs

```diff
@@ -18,14 +18,19 @@ public override void VisitManagedType (TypeDefinition type)
 				return;
 			
 			var name = type.Name;
-			TypeDefinition td;
 			// e.g. WatchKit.WKErrorCode and WebKit.WKErrorCode :-(
-			if (!enums.TryGetValue (name, out td))
+			if (!enums.TryGetValue (name, out var td))
 				enums.Add (name, type);
-			else if (td.Namespace.StartsWith ("OpenTK.", StringComparison.Ordinal)) {
-				// OpenTK duplicate a lots of enums between it's versions
-			} else {
-				Console.WriteLine ("!duplicate-type-name! {0} enum exists as both {1} and {2}", name, type.FullName, td.FullName);
+			else {
+				var (t1, t2) = Helpers.Sort (type, td);
+				if (t1.Namespace.StartsWith ("OpenTK.", StringComparison.Ordinal)) {
+					// OpenTK duplicate a lots of enums between it's versions
+				} else if (t1.IsNotPublic && String.IsNullOrEmpty (t1.Namespace)) {
```

Comment:
Question here, in the first if, we are doing a t1.Namespace.StartsWith and in the second one a String.IsNullOrEmpty (t1.Namespace) does that mean that t1 can be null? If that is the case, I would do the String.IsNullOrEmpty before the StartsWith to ensure we do not have null exceptions. Even more, moving the NullOrEmpty reduces the number of checks (not that we are very worried about performance in this code, but it never hurts).

Reply:
> does that mean that t1 can be null
No, `t1` can't be null there - `Sort` would throw;
However `t1,Namespace` can be (they are optional)

Note that this code is copied from master and the changes to the tools won't be merged (because some stuff needed to be disabled)